### PR TITLE
Convert blockcore to dotnet 3.1

### PR DESCRIPTION
**Pull request by xandronus**
Early prelook at dotnet 3.1 upgrade for early review

Both StratisD and BitcoinD start and start to sync. All tests build but haven't had a chance to run them yet.

From the Stratis PR looks like RPC project will need a bit more love. I will investigate that as well as I haven't tried RPC yet.

### CODE REVIEW

**Comment by dangershony** [comment]
I will help with testing

**Comment by fenix2222** [comment]
Any update on this? I can help

**Comment by dangershony** [comment]
@fenix2222 any help is most welcome

**Comment by xandronus** [comment]
@fenix2222  happy for your guidance. Do you know if we will need to make changes to get the RPC to work? I will try to have another pass at this on the weekend. I just got distracted with some other stuff

**Comment by xandronus** [comment]
image

Unit test status. Looking into integration tests seems like it might be something simple:
---- System.TypeLoadException : Could not load type 'Microsoft.Extensions.DependencyInjection.MvcJsonMvcCoreBuilderExtensions' from assembly 'Microsoft.AspNetCore.Mvc.Formatters.Json, Version=3.1.0.0

**Comment by xandronus** [comment]
Yeah RPC blows up, I think that's causing the integration test failures as well.

**Comment by xandronus** [comment]
Fixed the RPC. Node now starts up with RPC no problems. It also fixed all the broken integration tests

image

**Comment by xandronus** [comment]
#39

**Comment by dangershony** [comment]
@xandronus (and @fenix2222 as part of straits) good job 
I will try to review asap and If no one synced networks I will sync BTC and STRAT as soon as I can 

Anyone else that wants to help test this PR is most welcome as its a big change in code

Once we did some more testing I'll make this commit.

File: src/Stratis.Bitcoin.Tests/BlockPulling/BlockPullerBehaviorTests.cs [hunk 1]
```diff
@@ -21,8 +21,7 @@ public BlockPullerBehaviorTests()
             var ibdState = new Mock<IInitialBlockDownloadState>();
             ibdState.Setup(x => x.IsInitialBlockDownload()).Returns(() => true);
 
-            var loggerFactory = new ExtendedLoggerFactory();
-            loggerFactory.AddConsoleWithFilters();
+            var loggerFactory = ExtendedLoggerFactory.Create();
```
Comment (dangershony) [inline src/Stratis.Bitcoin.Tests/BlockPulling/BlockPullerBehaviorTests.cs]: I would actually love to get rid of this hack `ExtendedLoggerFactory` we just drop prefix stuff from logs it's not so crucial anyway

**Comment by dangershony** [comment]
I am having trouble building this locally oddly enough `Stratis.Bitcoin.Features.RPC.Tests` doesn't compile any ideas?

**Comment by xandronus** [comment]
> I am having trouble building this locally oddly enough `Stratis.Bitcoin.Features.RPC.Tests` doesn't compile any ideas?

Yeah I guess I missed that one - fixing it up right now. Can't believe I missed it!

**Comment by dangershony** [comment]
I think this can now go in to master

image

**Comment by DennisAMenace** [comment]
We will probably want to update the GitHub action scripts in this PR to reflect dotnet version 3.1 instead of 2.2 as this was pushed in https://github.com/block-core/blockcore/pull/43 before this PR could be added

**Comment by dangershony** [comment]
I think there is no container actions for 3.1 yet

**Pull request merged**